python/kserve/kserve/storage/storage.py

```diff
@@ -176,6 +176,17 @@ def get_S3_config():
         if accelerate:
             c = c.merge(Config(s3={"use_accelerate_endpoint": accelerate}))
 
+        # NOTE: If endpoint_url provided is legacy ("https://s3.amazonaws.com") and region is not global (us-east-1), set to virtual addressing style
+        # So that request would not return PermanentRedirect due to region not in the endpoint url
+        # AWS SDK retries under the hood to set the correct region when the valid virtual addressing style endpoint url is provided
+        endpoint_url = os.getenv("AWS_ENDPOINT_URL")
+        region = os.getenv("AWS_DEFAULT_REGION")
+        if endpoint_url == "https://s3.amazonaws.com" and region not in (
+            None,
+            "us-east-1",
```

Comment:
Why is "us-east-1" listed here, can it be handled by the SDK as well?

Reply:
Only buckets outside of `us-east-1` would need to add the region in the endpoint url (done so by virtual addressing style in SDK). It is listed here so it wouldn't affect the original logic for default regions